test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -148,6 +148,66 @@ public class HubConnectionHandlerTests
             }
         }
 
+        [Fact]
+        public async Task OberserverDoestThrowWhenOnNextIsCalledAfterChannelIsCompleted()
+        {
+            var observable = new Observable<int>();
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(s => s.AddSingleton(observable));
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<ObservableHub>>();
+
+            var waitForSubscribe = new TaskCompletionSource<object>();
+            observable.OnSubscribe = o =>
+            {
+                waitForSubscribe.TrySetResult(null);
+            };
+
+            var waitForDispose = new TaskCompletionSource<object>();
+            observable.OnDispose = o =>
+            {
+                waitForDispose.TrySetResult(null);
+            };
+
+            using (var client = new TestClient())
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+
+                async Task Subscribe()
+                {
+                    var results = await client.StreamAsync(nameof(ObservableHub.Subscribe));
+
+                    var items = results.OfType<StreamItemMessage>().ToList();
+
+                    // Ensuring that only the call before OnCompleted actually did anything.
+                    Assert.Single(items);
+                    Assert.Equal(1, (long)items[0].Item);
+                }
+
+                var subscribeTask = Subscribe();
+
+                await waitForSubscribe.Task.OrTimeout();
+
+                Assert.Single(observable.Observers);
+
+                observable.OnNext(1);
+
+                // Calling OnComplete to complete the observer. Further calls to OnNext should no-op
+                observable.Observers[0].OnCompleted();
```

Comment:
I think we want to kill the client connection, then trigger OnNext

Reply:
If we kill the client connection here then we can't verify the other call to OnNext succeeded since we wait on it later on. 
I don't think this is the prettiest but it covers the exact case that we want

Reply:
I don’t think it’s testing the right thing. In this scenario the client disconnected and the producer is happily calling OnNext